DSLink.Tests/NodeTests.cs

```diff
@@ -0,0 +1,239 @@
+            _mockResponder = new Mock<Responder>();
+            _mockSubManager = new Mock<SubscriptionManager>(_mockContainer.Object);
+            _superRootNode = new Node("", null, _mockContainer.Object);
+
+            _mockContainer.SetupGet(c => c.Connector).Returns(_mockConnector.Object);
+            _mockContainer.SetupGet(c => c.Responder).Returns(_mockResponder.Object);
+            _mockResponder.SetupGet(r => r.SuperRoot).Returns(_superRootNode);
+            _mockResponder.SetupGet(r => r.SubscriptionManager).Returns(_mockSubManager.Object);
+
+            _mockConnector.Setup(c => c.AddValueUpdateResponse(It.IsAny<JToken>()))
+                .Returns(Task.FromResult(false));
+        }
+
+        [Test]
+        public void TestBannedCharacters()
+        {
+            var bannedChars = Node.BannedChars;
+            foreach (char c in bannedChars)
+            {
+                Assert.Throws<ArgumentException>(() =>
+                {
+                    _superRootNode.CreateChild($"Test{c}").BuildNode();
+                });
+            }
+
+            var multiCharTest = new string(Node.BannedChars);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                _superRootNode.CreateChild(multiCharTest).BuildNode();
+            });
+
+            var noCharTest = "TestNoBannedChars";
+            Assert.DoesNotThrow(() =>
+            {
+                _superRootNode.CreateChild(noCharTest).BuildNode();
+            });
+        }
+
+        [Test]
+        public void TestCreateChildren()
+        {
+            int first = 5; // Number of children of root to create
+            int second = 100; // Number of children to make below each first node.
+            for (int i = 0; i < first; i++)
+            {
+                var firstNode = _superRootNode.CreateChild($"Node{i}").BuildNode();
+                for (int j = 0; j < second; j++)
+                {
+                    firstNode.CreateChild($"Node{j}").BuildNode();
+                }
+            }
+
+            Assert.AreEqual(first, _superRootNode.Children.Count);
+            foreach (var kv in _superRootNode.Children)
+            {
+                Assert.AreEqual(second, kv.Value.Children.Count);
+            }
+        }
+
+        [Test]
+        public void TestRemoveChildren()
+        {
+            const int childrenCount = 100;
+
+            Assert.AreEqual(0, _superRootNode.Children.Count);
+
+            for (int i = 0; i < childrenCount; i++)
+            {
+                _superRootNode.CreateChild($"Node{i}").BuildNode();
+            }
+
+            Assert.AreEqual(childrenCount, _superRootNode.Children.Count);
+
+            for (int i = 0; i < childrenCount; i++)
+            {
+                _superRootNode[$"Node{i}"].RemoveFromParent();
+            }
+
+            Assert.AreEqual(0, _superRootNode.Children.Count);
+        }
+
+        [Test]
+        public void TestSubscriberUpdate()
```

Comment:
For tests involving interactions testing, I usually like to write the verified interaction in the test name. In fact, I'd rename many of your tests to have a clearer name regarding their intent.